extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -219,12 +219,14 @@ export abstract class AzureAuth implements vscode.Disposable {
 					console.log('Base token was empty, account is stale.');
 					return undefined;
 				}
+
 				try {
 					await this.refreshAccessToken(account.key, baseToken.refreshToken, tenant, resource);
 				} catch (ex) {
-					console.log(ex);
-					account.isStale = true;
-					return undefined;
+					console.log(`Could not refresh access token for ${JSON.stringify(tenant)} - silently removing the tenant from the user's account.`);
```

Comment:
What about other general errors that occur while trying to refresh? Is the error not something we can check for more specifically? 

It might be worth thinking about asking the user whether they want us to remove the tenant so that at least they're aware it's happening. (I'm a little worried most users won't know what that means though so not sure what the best approach is here)

Reply:
Synced offline

Reply:
Talked offline - going to get this in and then if users have an issue with missing resources can look into exposing this more publically to them. But it should be enough of an edge case that this is fine to just do and log in the console.